pkg/volume/persistent_claim/persistent_claim.go

```diff
@@ -55,30 +56,39 @@ func (plugin *persistentClaimPlugin) CanSupport(spec *volume.Spec) bool {
 	return spec.Volume != nil && spec.Volume.PersistentVolumeClaim != nil
 }
 
-func (plugin *persistentClaimPlugin) NewMounter(spec *volume.Spec, pod *api.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *persistentClaimPlugin) getPersistentVolumeBySpec(spec *volume.Spec, pod *api.Pod) (*api.PersistentVolumeClaim, *api.PersistentVolume, error) {
```

Comment:
With the volume manager you will need to something similar to what I'm doing in the attach/detach controller and dereference PVCs as soon as you encounter them because you'll have to extract the pluginName/diskName to operate on--the volume manager can't operate on the PVC name since the underlying disk may already be attached directly or via another PVC with a different name. That means you should always dereference PVCs as soon as you encounter them and always operate on the VolumeSpec. If you do this, then you can remove the `Mounter` interfaces altogether from `persistent_claim/persistent_claim.go`